ccl/schemachangerccl: skip TestBackupMixedVersionElements_base_drop_m…

herkolategan opened the pull request:
…ultiple_columns_separate_statements

This skip also adds code to skip datadriven tests in `sctest`.

Refs: #103896
Reason: flaky test

Release note: None
Epic: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/104864)
<!-- Reviewable:end -->

tbg commented:
[TestBackupMixedVersionElements_base_add_column_with_stored](https://teamcity.cockroachdb.com/viewLog.html?buildId=10563281&tab=buildResultsDiv&buildTypeId=Cockroach_Ci_TestsGcpLinuxX8664BigVm_CclUnitTests#testNameId8387439225146341446) also failed for me in https://teamcity.cockroachdb.com/viewLog.html?buildId=10563281&tab=buildResultsDiv&buildTypeId=Cockroach_Ci_TestsGcpLinuxX8664BigVm_CclUnitTests (https://github.com/cockroachdb/cockroach/pull/104945#issuecomment-1594745925)

Maybe you want to add it too. It doesn't have a GH issue yet, either.

rafiss reviewed (changes requested):
I think this way of skipping is a bit hard to figure out. Or at least, it's different from how we normally do skips for datadriven tests. Could you change it to something like this?

```
diff --git a/pkg/sql/schemachanger/sctest/cumulative.go b/pkg/sql/schemachanger/sctest/cumulative.go
index b8cc2e5648f..23f1561834b 100644
--- a/pkg/sql/schemachanger/sctest/cumulative.go
+++ b/pkg/sql/schemachanger/sctest/cumulative.go
@@ -566,6 +566,12 @@ func cumulativeTest(
 		require.Zero(t, numTestStmts, "only one test command per-test, "+
 			"and it must be the last one.")
 		switch d.Cmd {
+		case "skip":
+			issueNum, err := strconv.Atoi(d.CmdArgs[0].Key)
+			if err != nil {
+				t.Fatal(err)
+			}
+			skip.WithIssue(t, issueNum)
 		case "setup":
 			// Store setup stmts into `setup` slice (without executing them).
 			stmts, err := parser.Parse(d.Input)
diff --git a/pkg/sql/schemachanger/testdata/end_to_end/drop_multiple_columns_separate_statements b/pkg/sql/schemachanger/testdata/end_to_end/drop_multiple_columns_separate_statements
index 7610a9904d9..4fa037a6dc0 100644
--- a/pkg/sql/schemachanger/testdata/end_to_end/drop_multiple_columns_separate_statements
+++ b/pkg/sql/schemachanger/testdata/end_to_end/drop_multiple_columns_separate_statements
@@ -1,3 +1,6 @@
+skip 103896
+----
+
 # As a part of these tests we will have cascaded drops
 # which will clean up secondary indexes, computed columns
 # since j and k are being dropped. We will concurrently
```

herkolategan committed:
ccl/schemachangerccl: skip TestBackupMixedVersionElements_base_drop_multiple_columns_separate_statements

This skip also adds code to skip datadriven tests in `sctest`.

Refs: #103896
Reason: flaky test

Release note: None
Epic: None

herkolategan reviewed (commented):
Thanks @tbg I'll create an issue & skip for that.
@rafiss Right, thanks for pointing that out, it will probably make it clearer in the logs as well.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/104864#-:-NY3sYZt8wt345yvopcs:b-sq84us)* status: :shipit: complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

rafiss reviewed (approved):
lgtm! thanks

herkolategan commented:
TFTR!
bors r=rafiss
